include/deal.II/non_matching/mapping_info.h

```diff
@@ -855,7 +869,7 @@ namespace NonMatching
     const typename Triangulation<dim, spacedim>::cell_iterator &cell,
     const std::vector<Point<dim>>                              &unit_points_in)
   {
-    reinit(cell, Quadrature<dim>(unit_points_in));
+    reinit(cell, make_array_view(unit_points_in));
```

Comment:
Just for completeness, one could also have addressed this by overloading `reinit()` with a version that takes an rvalue reference as second argument and simply moves the object.

I do think that the solution you chose is the more useful one in practice.